src/rf/apps/workers/process.py

```diff
@@ -58,48 +63,84 @@ def start_polling(self):
                         delete_message = self.check_timeout(record['data'])
                     elif job_type in JOB_VALIDATE:
                         if record['eventSource'] == 'aws:s3':
-                            delete_message = self.validate_image(record)
+                            delete_message = self.__validate_image(record)
 
                 if delete_message:
                     self.q.delete_message(message)
 
-    def validate_image(self, data):
+    def __validate_image(self, data):
         """
         Use Gdal to verify an image is properly formatted and can be further
         processed.
         data -- attribute data from SQS.
         """
-        # Pass image to Gdal to verify.
-        # If it succeeds continue else return False.
-
-        key = data['s3']['object']['key']
-        payload_uuid = self.extract_uuid_from_aws_key(key)
-
-        url = self.make_s3_url(data['s3']['bucket']['name'], key)
 
-        # TODO - Right now we don't save data to the DB with upload. So hard
-        # code something for testing.
-        try:
-            image = LayerImage.objects.get(s3_uuid=payload_uuid)
+        def ensure_band_count(key_string, range=None):
+            """
+            Gets the first `range` bytes from the s3 resource and uses it to
+            determine the number of bands in the image.
+            """
+            connection = S3Connection()
+            bucket = connection.get_bucket(settings.AWS_BUCKET_NAME)
+            s3_key = Key(bucket)
+            s3_key.key = key_string
+            random_filename = '/tmp/' + str(uuid.uuid4())
+            tempfile = open(random_filename, 'w')
+            if range is not None:
+                range = {'Range': 'bytes=' + range}
+                s3_key.get_contents_to_file(tempfile, headers=range)
+            else:
+                s3_key.get_contents_to_file(tempfile)
+
+            tempfile.close()
+            try:
+                validator = gdal.Open(random_filename)
```

Comment:
Does this gdal binding support [vsicurl](https://trac.osgeo.org/gdal/wiki/UserDocs/ReadInZip#vsicurl-toreadfromHTTPorFTPfilespartialdownloading)? If so, we could avoid downloading the file altogether.